clippy_lints/src/casts/mod.rs

```diff
@@ -506,6 +507,30 @@ declare_clippy_lint! {
     "casting the result of `abs()` to an unsigned integer can panic"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for a raw slice being cast to a slice pointer.
+    ///
+    /// ### Why is this bad?
+    /// This can result in multiple `&mut` references to the same location when only a pointer is
+    /// required.
```

Comment:
I think it would be worth expanding on this part, maybe mentioning that `ptr::slice_from_raw_parts` is a safe function and so you don't have to uphold the safety requirements of `slice::from_raw_parts` + a link to https://doc.rust-lang.org/std/slice/fn.from_raw_parts.html#safety